UnicodeJsps/src/main/java/org/unicode/jsp/UnicodeUtilities.java

```diff
@@ -874,25 +895,32 @@ private static String getName(
         return result.toString();
     }
 
-    private static String getHex(int codePoint, boolean ucdFormat) {
+    private static String getHex(
+            int codePoint, boolean ucdFormat, List<String> additionalParameters) {
         String hex = com.ibm.icu.impl.Utility.hex(codePoint, 4);
         final String string =
                 "<code>"
-                        + ("<a target='c' href='character.jsp?a=" + hex + "'>")
+                        + ("<a target='c' href='character.jsp?a="
+                                + hex
+                                + additionalParameters.stream()
+                                        .map(p -> "&" + p)
+                                        .collect(Collectors.joining())
+                                + "'>")
                         + (ucdFormat ? "" : "U+")
                         + hex
                         + "</a></code>";
         return string;
     }
 
-    private static String getHex(String string, String separator, boolean ucdFormat) {
+    private static String getHex(
+            String string, String separator, boolean ucdFormat, List<String> additionalParameters) {
         StringBuilder result = new StringBuilder();
         int cp;
         for (int i = 0; i < string.length(); i += UTF16.getCharCount(cp)) {
```

Comment:
FYI: At some point, we could change this kind of code to use [CharSequence.codePoints()](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/CharSequence.html#codePoints()).

Reply:
Yes.